Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +27,51 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return totalAmpHour_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+
+    if(current_<0){
+       return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
-{
-    return QTime::currentTime();
+{    QTime charging;
+     charging.setHMS(0,0,0,0);
+
+    if(isCharging()){
+    charging= charging.addMSecs(qAbs(totalAmpHour_/current_*MS_TO_HOURS));
+
+     return charging;
+    }
+
+   charging = charging.addMSecs(qAbs(( BATTERY_AMP_HOUR_CAPACITY-totalAmpHour_)/current_*MS_TO_HOURS));
+    return charging;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
     Q_UNUSED(batteryData);
```

Comment:
This macro is here to handle unused arguments, since it's being use, we can remove this!